Automattic/wp-calypso
# ESLint: Fix common type errors caused by wrong capitalization

While working on Calypso during the HACK week, I noticed that in many of the files we're using an incorrect capitalization for types - for example `String` instead of `string`. This was causing a lot of ESLint warnings, and made working on such files a bit of pain for the developer, as they should fix such errors too.

This PR fixes many of those type ESLint warnings that occur often. There are surely some that remain, but they will be much less after this PR is merged. 

This PR will hopefully remove some of the developer frustration while working on existing files in Calypso, as it removes almost 6000 ESLint warnings.

Once we merge this, I'm planning to do some more updates that bulk-reduce ESLint warnings.

#### Changes proposed in this Pull Request

* ESLint: Fix common type errors caused by wrong capitalization

#### Testing instructions

* No testing should be necessary - changes are only in code comments, JSDoc and READMEs. Just verify that changes make sense.
